File: ToDo/ToDo.Web/Modules/ServerTypes/Administration/PersonColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { Gender } from "../Modules/Administration.Person.Gender";
import { PersonRow } from "./PersonRow";

export interface PersonColumns {
    PersonId: Column<PersonRow>;
    FirstName: Column<PersonRow>;
    LastName: Column<PersonRow>;
    PrimaryImage: Column<PersonRow>;
    GalleryImages: Column<PersonRow>;
    BirthDate: Column<PersonRow>;
    BirthPlace: Column<PersonRow>;
    Gender: Column<PersonRow>;
    Height: Column<PersonRow>;
}

export class PersonColumns extends ColumnsBase<PersonRow> {
    static readonly columnsKey = 'Administration.Person';
    static readonly Fields = fieldsProxy<PersonColumns>();
}

[Gender]; // referenced types

File: ToDo/ToDo.Web/Modules/Administration/Person/PersonColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;
using ToDo.Modules.Administration.Person;

namespace ToDo.Administration.Columns;

[ColumnsScript("Administration.Person")]
[BasedOnRow(typeof(PersonRow), CheckNames = true)]
public class PersonColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int PersonId { get; set; }
    [EditLink]
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string PrimaryImage { get; set; }
    public string GalleryImages { get; set; }
    public DateTime BirthDate { get; set; }
    public string BirthPlace { get; set; }
    public Gender Gender { get; set; }
    public int Height { get; set; }
}